packages/shared_preferences/shared_preferences_macos/macos/Classes/SharedPreferencesPlugin.swift

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FlutterMacOS
+import Foundation
+
+public class SharedPreferencesPlugin: NSObject, FlutterPlugin {
```

Comment:
It feel inefficient to me that we now have a Swift and Objective-C implementations that are doing the exact same thing. Could we share code?

Reply:
/cc @amirh

Reply:
A Pod can't reference source outside its directory, so I don't think there's any way to actually use the same code for both unless we taught `flutter` to do something special just for this case (e.g., a pubspec option to say what subdirectory contains the implementation instead of it being hard-coded).

If there were non-trivial logic we could share it by extracting it to a Pod that both depended on, but in this case the nontrivial logic is just `NSUserDefaults`.